zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/deployment/DbResourceState.java

```diff
@@ -0,0 +1,315 @@
+          DbForeignKey<DbTenantAwareKey<DbLong>>>
+      resourceKeyByResourceIdAndVersionTagColumnFamily;
+
+  private final LoadingCache<TenantIdAndResourceId, PersistedResource>
+      resourcesByTenantIdAndIdCache;
+
+  public DbResourceState(
+      final ZeebeDb<ZbColumnFamilies> zeebeDb,
+      final TransactionContext transactionContext,
+      final EngineConfiguration config) {
+    tenantIdKey = new DbString();
+    dbResourceKey = new DbLong();
+    tenantAwareResourceKey =
+        new DbTenantAwareKey<>(tenantIdKey, dbResourceKey, PlacementType.PREFIX);
+    fkResourceKey = new DbForeignKey<>(tenantAwareResourceKey, ZbColumnFamilies.RESOURCES);
+    dbPersistedResource = new PersistedResource();
+    resourcesByKey =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.RESOURCES,
+            transactionContext,
+            tenantAwareResourceKey,
+            dbPersistedResource);
+
+    dbResourceId = new DbString();
+    resourceVersion = new DbLong();
+    idAndVersionKey = new DbCompositeKey<>(dbResourceId, resourceVersion);
+    tenantAwareIdAndVersionKey =
+        new DbTenantAwareKey<>(tenantIdKey, idAndVersionKey, PlacementType.PREFIX);
+    resourceByIdAndVersionColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.RESOURCE_BY_ID_AND_VERSION,
+            transactionContext,
+            tenantAwareIdAndVersionKey,
+            fkResourceKey);
+
+    dbDeploymentKey = new DbLong();
+    tenantAwareResourceIdAndDeploymentKey =
+        new DbTenantAwareKey<>(
+            tenantIdKey, new DbCompositeKey<>(dbResourceId, dbDeploymentKey), PlacementType.PREFIX);
+    resourceKeyByResourceIdAndDeploymentKeyColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.RESOURCE_KEY_BY_RESOURCE_ID_AND_DEPLOYMENT_KEY,
+            transactionContext,
+            tenantAwareResourceIdAndDeploymentKey,
+            fkResourceKey);
+
+    dbVersionTag = new DbString();
+    tenantAwareResourceIdAndVersionTagKey =
+        new DbTenantAwareKey<>(
+            tenantIdKey, new DbCompositeKey<>(dbResourceId, dbVersionTag), PlacementType.PREFIX);
+    resourceKeyByResourceIdAndVersionTagColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.RESOURCE_KEY_BY_RESOURCE_ID_AND_VERSION_TAG,
+            transactionContext,
+            tenantAwareResourceIdAndVersionTagKey,
+            fkResourceKey);
+
+    versionManager =
+        new VersionManager(
+            DEFAULT_VERSION_VALUE, zeebeDb, ZbColumnFamilies.RESOURCE_VERSION, transactionContext);
+
+    resourcesByTenantIdAndIdCache =
+        CacheBuilder.newBuilder()
+            .maximumSize(config.getResourceCacheCapacity())
+            .build(
+                new CacheLoader<>() {
+                  @Override
+                  public PersistedResource load(final TenantIdAndResourceId key) {
+                    Optional<PersistedResource> latestResourceById =
```

Comment:
🔧 Seems like the code style checker is complaining that this is not `final`. You'll need to fix it before the PR can be  merged.

Reply:
You can see the log here: https://github.com/camunda/camunda/actions/runs/12070306563/job/33659607803?pr=24775